src/logic/safe/transactions/gas.ts

```diff
@@ -93,19 +93,40 @@ export const estimateTxGasCosts = async (
   }
 }
 
-// Parses the result of OpenEthereum/Parity and Nethermind error messages and returns the value
-export const getNonGETHErrorDataResult = (errorMessage: string): string | undefined => {
+// Parses the result from the error message (GETH, OpenEthereum/Parity and Nethermind) and returns the data value
+export const getDataFromNodeErrorMessage = (errorMessage: string): string | undefined => {
   // Extracts JSON object from the error message
   const [, ...error] = errorMessage.split('\n')
   try {
     const errorAsJSON = JSON.parse(error.join(''))
 
+    // For new GETH nodes they will return the data as error in the format:
+    // {
+    //   "originalError": {
+    //     "code": number,
+    //     "data": string,
+    //     "message": "execution reverted: ..."
+    //   }
+    // }
+    if (errorAsJSON.originalError && errorAsJSON.originalError.data) {
+      return errorAsJSON.originalError.data
+    }
+
+    // OpenEthereum/Parity nodes will return the data as error in the format:
+    // {
+    //     "error": {
+    //         "code": number,
+    //         "message": string,
+    //         "data": "revert: 0x..." -> this is the result data that should be extracted from the message
+    //      },
+    //     "id": number
+    // }
     if (errorAsJSON?.data) {
       const [, dataResult] = errorAsJSON.data.split(' ')
       return dataResult
     }
   } catch (error) {
-    console.error(`Error trying to extract data from openEthereum/Nethermind error message: ${errorMessage}`)
+    console.error(`Error trying to extract data as error from error message: ${errorMessage}`)
```

Comment:
I would rewritte it as
`Error trying to extract error data from error message:`